src/formatDate.js

```diff
@@ -8,7 +8,33 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatCopy = [...fromFormat];
+  const toFormatCopy = [...toFormat];
+  const deleteLastElement = fromFormatCopy.pop();
+  const partOfDate = date.split(deleteLastElement);
+  const partsOfDate = {};
+
+  fromFormatCopy.forEach((part, index) => {
+    partsOfDate[part] = partOfDate[index];
+  });
+
+  if (toFormatCopy.includes('YY') && partsOfDate['YYYY']) {
+    partsOfDate['YY'] = partsOfDate['YYYY'].slice(-2);
+  }
```

Comment:
The conversion from 'YYYY' to 'YY' assumes that 'YYYY' is always present, which might not be the case. Ensure that the logic handles cases where 'YYYY' is not provided in `fromFormat`.